### PR TITLE
configurable pubsub signing

### PR DESCRIPTION
**Pull request by Stebalien**
I'd like to sneak this into the release so we can turn on strict verification ASAP.

Depends on https://github.com/ipfs/go-ipfs-config/pull/18 and a gx update.

### CODE REVIEW

File: core/core.go [hunk 1]
```diff
@@ -475,14 +475,23 @@ func (n *IpfsNode) startOnlineServicesWithHost(ctx context.Context, host p2phost
 
 		var service *pubsub.PubSub
 
+		var pubsubOptions []pubsub.Option
```
Comment (schomatis) [inline core/core.go]: What happens if they are both set?

File: test/sharness/t0180-pubsub.sh [hunk 2]
```diff
@@ -10,83 +10,143 @@ test_expect_success 'init iptb' '
...
+  test_expect_success 'pubsub' '
+    echo "testOK" > expected &&
+    touch empty &&
+    mkfifo wait ||
+    test_fsh echo init fail
+  
+    # ipfs pubsub sub is long-running so we need to start it in the background and
+    # wait put its output somewhere where we can access it
+    (
+      ipfsi 0 pubsub sub --enc=ndpayload testTopic | if read line; then
```
Comment (schomatis) [inline test/sharness/t0180-pubsub.sh]: What is `ipfsi`? (I mean, where can I read some information about it?)

Comment (Stebalien) [reply test/sharness/t0180-pubsub.sh]: It's a function defined in `test/sharness/lib/iptb-lib.sh`.

File: core/core.go (hunk 1 above)
Comment (Stebalien) [reply core/core.go]: The user won't be able to send any messages. We should error in that case, shouldn't we...

Comment (Stebalien) [reply core/core.go]: https://github.com/libp2p/go-libp2p-pubsub/pull/122

Comment (schomatis) [reply core/core.go]: Re-reading this now with some more context it seems like a valid case (although I'm not sure how useful it is), that is, we are requiring signed messages but we don't sign them ourselves.

Comment (Stebalien) [reply core/core.go]: It is a valid case, technically. But it's so useless that we should probably get rid of it.

**Comment by Stebalien** [comment]
> it's a string, but does it carry some special meaning?

It's an arbitrary (binary) string. At one point, it was the hash of a topic descriptor but that never really got finished.

**Comment by Stebalien** [comment]
@schomatis mind ticking https://github.com/ipfs/go-ipfs-config/pull/18 as well?

File: core/core.go (hunk 1 above)
Comment (Stebalien) [reply core/core.go]: (actually, I may just do that in a followup patch once someone takes a look at that PR)

Comment (vyzo) [reply core/core.go]: if strict signing is set but signing is disabled, then it will be ignored.

Comment (Stebalien) [reply core/core.go]: Got it. Now that the option validation patch has been merged, it should return an error on construction. That should be the least surprising result.

**Pull request merged**